Library/Homebrew/download_strategy.rb

```diff
@@ -713,6 +713,20 @@ def initialize(name, resource)
     end
   end
 
+  def source_modified_time
+    # Look for the file timestamps under {#cached_location} first because
+    # newly-unpacked directory can have timestamps of the moment of copying.
+    # Filter CVS's files because the timestamp for each of them is the moment
+    # of clone.
+    if cached_location.exist?
+      cached_location.to_enum(:find).select do |f|
+        !f.to_s.include?("CVS/") && f.file?
+      end.map(&:mtime).max
```

Comment:
I would suggest to avoid `select` but use `find` directly to gain better performance. Such as
```ruby
max_mtime = Time.at(0)
cached_location.find do |f|
  if f.directory?
    if f.basename.to_s == "CVS"
      Find.prune
    else
      next
    end
  end
  mtime = f.mtime
  max_mtime = mtime if mtime > max_mtime
end
max_mtime
```